### PR TITLE
chore(geo): update configs for newer scoped configurations and providers - Geo milestone 1 - PR 7

### PR DESCRIPTION
**Pull request by TreTuna**
#### Description of changes

Update config methods to be more agnostic and scope configs to providers. The CLI's exported config to `aws-exports` was updated to nest `amazon_location_services` config under the `geo` key (before it was not scoped and just under `geo`). This updates Geo's config to handle that scoping. Also, updates to allow for more scoped provider configurations in the future, making it more agnostic to the provider.

#### Description of how you validated changes
Updated tests and ran this through my example application

image

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)
_Documentation PRs are next in line_

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/geo/src/Geo.ts [hunk 1]
```diff
@@ -79,7 +76,7 @@ export class GeoClass {
 		);
 		if (pluggable === undefined) {
 			logger.debug('No plugin found with providerName', providerName);
-			return null;
+			throw 'No plugin found in Geo for the provider';
```
Comment (jamesaucode) [inline packages/geo/src/Geo.ts:79]: I prefer if we throw a `new Error` instead of just string

File: packages/geo/src/Geo.ts [hunk 2]
```diff
@@ -141,47 +131,34 @@ export class GeoClass {
...
-			return Promise.reject('No plugin found in Geo for the provider');
-		}
+		const prov = this.getPluggable(providerName);
 
-		const responsePromise = await prov.searchByText(text, options);
-		return responsePromise;
+		try {
+			return await prov.searchByText(text, options);
+		} catch (error) {
+			throw new Error(error);
```
Comment (jamesaucode) [inline packages/geo/src/Geo.ts:146]: Why do we catch the errors if we just throw them back immediately?

**Review by jamesaucode — COMMENTED** [review]
Looks good! Just a couple nit picks

File: packages/core/src/Parser.ts [hunk 3]
```diff
@@ -55,12 +55,9 @@ export const parseMobileHubConfig = (config): AmplifyConfig => {
...
 	if (config['geo']) {
-		amplifyConfig.Geo = Object.assign({}, config.geo);
-	}
-
-	// Geo
-	if (config['geo']) {
-		amplifyConfig.Geo = Object.assign({}, config.geo);
+		amplifyConfig.Geo = {
+			AmazonLocationServices: config.geo['amazon_location_services'],
+		};
```
Comment (Amplifiyer) [inline packages/core/src/Parser.ts:60]: This should be part of the Geo category implementation. i.e. Geo category when added a provider names `AmazonLocationServices` should send the config for `amazon_location_services` to it. Parser shouldn't have provider information.

File: packages/geo/__tests__/Geo.test.ts [hunk 4]
```diff
@@ -133,7 +150,7 @@ describe('Geo', () => {
 			geo.configure(awsConfig);
 
 			const maps = [];
-			const availableMaps = awsConfig.geo.maps.items;
+			const availableMaps = awsConfig.geo.amazon_location_services.maps.items;
 			for (const mapName in availableMaps) {
```
Comment (Amplifiyer) [inline packages/geo/__tests__/Geo.test.ts:154]: Now with proper segmentation of Category and Providers, the tests should be updated as such. e.g. Geo category tests shouldn't rely on `amazon_location_services` and should just mock the provider implementation. Provider tests should test the implementation (parsing the config etc.)

**Review by Amplifiyer — COMMENTED** [review]
Just couple of structural nits. Non blocking.

File: packages/geo/src/Geo.ts (hunk 2 above)
Comment (TreTuna) [reply packages/geo/src/Geo.ts:146]: Because I forgot to add the `logger.debug(error)` statement that should also be here. 🤦

File: packages/geo/src/Geo.ts (hunk 1 above)
Comment (TreTuna) [reply packages/geo/src/Geo.ts:79]: Yeah, makes sense. 👍

File: packages/core/src/Parser.ts (hunk 3 above)
Comment (TreTuna) [reply packages/core/src/Parser.ts:60]: Spoke with Praveen offline about this. We are going to keep this as is for now and update it after the initial developer preview release.

File: packages/geo/__tests__/Geo.test.ts (hunk 4 above)
Comment (TreTuna) [reply packages/geo/__tests__/Geo.test.ts:154]: Spoke with Praveen offline about this. We are going to keep this as is for now and update it after the initial developer preview release.

File: packages/geo/src/Geo.ts [hunk 5]
```diff
@@ -41,11 +41,6 @@ export class GeoClass {
 		this._config = {};
 		this._pluggables = [];
 		logger.debug('Geo Options', this._config);
-
-		this.getAvailableMaps.bind(this);
-		this.getDefaultMap.bind(this);
-		this.searchByText.bind(this);
-		this.searchByCoordinates.bind(this);
 	}
```
Comment (wlee221) [inline packages/geo/src/Geo.ts:44]: Why were these removed?

**Review by wlee221 — APPROVED** [review]
LGTM, just got one question.

File: packages/geo/src/Geo.ts (hunk 5 above)
Comment (TreTuna) [reply packages/geo/src/Geo.ts:44]: They were unneeded. Tested it without them and everything works fine.
